ios/RNSScreenStack.mm

```diff
@@ -1200,6 +1230,21 @@ - (void)mountingTransactionDidMount:(const facebook::react::MountingTransaction
       break;
     }
   }
+
+  if (!self->_toBeDeletedScreens.empty()) {
+    __weak RNSScreenStackView *weakSelf = self;
+    // We want to run after container updates are performed (transitions etc.)
+    dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
How are you sure that it will run in the correct moment? Maybe it would be better to do it in some platform methods from `UINavigationController` or in `viewDidDisappear`? Not sure though.

Reply:
`viewDidDisappear` is called e.g. for views that are no longer on top of the stack (e.g. new screen has been pushed in), so it does not serve our need. 

Only method that would make sense here is `didMoveToParentViewController:` called with `nil`, however I'm not sure of its exact timing with respect to react methods & I believe (haven't tested it though), that it has the same drawback as `viewDidDisappear`. 

Only concern here is that the transitions started in `maybeAddToParentAndUpdateContainer` will break in result of `screenView._controller` being set to `nil` during or before the transition. It seems that this could happen, if the transitions:

1. were asynchronous,
2. held only weak reference to the `RNSScreen`'s used in the transition,
3. or referenced the `RNSScreen`'s only via `screenView->_controller`.

I assumed that 3rd point is unlikely, because `_controller` is our private field, and AFAIK (just checked in UIKit docs) `UIView` does not expose any accessor to its owning view controller in UIKit API, thus system does not do that. 
Our animation code uses only reversed reference: it obtains `RNSScreenView` instance  from `RNSScreen`, hence we're safe on this front. 

2nd point *seems* to be false, haven't found any information on this, saying only empirically.

1st point: 🤷🏻‍♂️ Not really sure how transitions work & haven't managed to find any information on threading etc. 

Summary: This surely can be implemented in other ways, however did not see a better approach at this moment, while the suggested one seems to solve the issue w/o introducing regressions (at least my testing suggests so).

Reply:
> viewDidDisappear is called e.g. for views that are no longer on top of the stack (e.g. new screen has been pushed in), so it does not serve our need.

I think we have some heuristics already for when it is the top screen that is dismissed but yeah, if it works then we can stick with that.